### PR TITLE
Yet another update on the documentation

### PR DESCRIPTION
**Pull request by HubLot**
Hi,

Here a big update on the documentation and how it's built.

First, I started to remove the folder `pbxplore/demo` which contained demo files we don't use anymore in the documentation. Now, the files are directly taken from the PDB or from the `demo_doc` folder.

The conversion of notebooks into rst and html files are now done with the [`nbsphinx` tool](https://nbsphinx.readthedocs.io/en/0.2.13/). It's a sphinx extension, meaning the conversion is now integrated in the `sphinx-build` command. We don't need anymore to first convert the notebooks in markdown then in rst. Hence, `doc/source/build.sh` and `doc/source/pandoc_fix_img.py` were removed and `doc/source/conf.py`has been updated.

The notebooks have been updated to use a python 3 kernel. A special cell has been added in the top of each notebook :

```
.. contents:: Table of Contents
   :local:
   :backlinks: none
```

It's a `Raw NBconvert` type cell which format is `reST`. This allows the nbsphinx extension to parse it as reST code in order to add a table of content for the notebook (which was done manually before).

Plus, the notebooks are still executed before converted to ensure the content is still valid. It's useful when testing the doc in Travis (it's a nbsphinx option in `doc/source/conf.py`).

Finally, the docs is only tested in python 3.5 in Travis.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,7 @@
 **Dev**
 - Update documentation
+- Notebooks doc are know Python3
```
Comment (jbarnoud) [inline CHANGELOG]: "now" instead of "know"

File: .gitignore [hunk 2]
```diff
@@ -15,9 +15,9 @@ test_tmp/
 # ignore macos files
 .DS_Store
 # ignore API doc generated files
-doc-api/build/*
-doc-api/source/notebooks/*.rst
-doc-api/source/notebooks/*_files
+doc/build/*
+doc/source/notebooks/*
+!doc/source/notebooks/*.ipynb
```
Comment (jbarnoud) [inline .gitignore]: What does the ! do?

File: CHANGELOG (hunk 1 above)
Comment (HubLot) [inline CHANGELOG]: arf....

File: .gitignore (hunk 2 above)
Comment (HubLot) [inline .gitignore]: "don't ignore". Useful when combined with the line above

Comment (pierrepo) [inline .gitignore]: interesting !

**Comment by pierrepo** [comment]
Thanks @HubLot. You're on fire!

**Pull request merged**